### PR TITLE
refactor(store): remove legacy space

### PR DESCRIPTION
**Pull request by Saul-Mirone**
Closes: BS-136

### CODE REVIEW

**Comment by Saul-Mirone** [comment]
* **#7559** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7559?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Saul-Mirone and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/framework/store/src/store/doc/block-collection.ts [hunk 1]
```diff
@@ -63,11 +61,58 @@ export class BlockCollection extends Space<FlatBlockMap> {
...
 
   private _shouldTransact = true;
 
+  protected readonly _yBlocks: Y.Map<YBlock>;
+
+  /**
+   * @internal Used for convenient access to the underlying Yjs map,
+   * can be used interchangeably with ySpace
+   */
+  protected readonly _ySpaceDoc: Y.Doc;
```
Comment (Flrande) [inline packages/framework/store/src/store/doc/block-collection.ts:108]: After removing the space, some names can also be improved.

File: packages/framework/store/src/store/store.ts [hunk 2]
```diff
@@ -167,11 +171,11 @@ export class Store {
     }
   }
 
-  addSpace(space: Space) {
-    this.spaces.set(space.id, space);
+  addSpace(blockCollection: BlockCollection) {
```
Comment (Flrande) [inline packages/framework/store/src/store/store.ts:174]: Same

**Review by Flrande — APPROVED** [review]

File: packages/framework/store/src/store/doc/block-collection.ts (hunk 1 above)
Comment (Saul-Mirone) [reply packages/framework/store/src/store/doc/block-collection.ts:108]: I'll update them and use signal in next pr.

File: packages/framework/store/src/store/store.ts (hunk 2 above)
Comment (Saul-Mirone) [reply packages/framework/store/src/store/store.ts:125]: Store will be removed in next pr.

**Pull request merged**